lib/raygun.ts

```diff
@@ -140,14 +141,70 @@ class Raygun {
     this._tags = tags;
   }
 
+  transport() {
+    if (this._batch && this._batchTransport) {
+      return this._batchTransport;
+    }
+
+    const client = this;
+
+    return {
+      send(message: string, callback: Function) {
+        const apiKey = client._apiKey;
+
+        if (!apiKey) {
+          console.error(
+            `Encountered an error sending an error to Raygun. No API key is configured, please ensure .init is called with api key. See docs for more info.`
+          );
+          return message;
+        }
+
+        debug(`sending message to raygun (${message.length} bytes)`);
+
+        function wrappedCallback<R>(error: Error | null, response: R) {
+          const [seconds, nanoseconds] = process.hrtime(startTime);
+          const durationInMs = Math.round(seconds / 1000 + nanoseconds / 1e6);
+          if (error) {
+            debug(
+              `error sending message (duration=${durationInMs}ms): ${error}`
+            );
+          } else {
+            debug(`successfully sent message (duration=${durationInMs}ms)`);
+          }
+          if (!callback) {
+            return;
+          }
+          if (callback.length > 1) {
+            return callback(error, response);
+          } else {
+            return callback(response);
+          }
+        }
+
+        const startTime = process.hrtime();
+        return raygunTransport.send({
+          message,
+          callback: wrappedCallback,
+          batch: false,
+          http: {
+            host: client._host,
+            port: client._port,
+            useSSL: !!client._useSSL,
+            apiKey,
+          },
+        });
```

Comment:
Might not be worth it but this looks similar to the batch transports send method so it could be good to consolidate these?

Reply:
Agree, there's a few details here that would be better not to duplicate. I've refactored the timing and callback logic so they're shared and it feels nicer.